src/NuGet.Core/NuGet.Versioning/VersionRange.cs

```diff
@@ -317,21 +317,28 @@ private static NuGetVersion GetNonSnapshotVersion(NuGetVersion version)
 
             var lastLabel = version.ReleaseLabels.LastOrDefault() ?? string.Empty;
 
-            if (lastLabel.EndsWith("-", StringComparison.Ordinal))
+            var endsWithZero = lastLabel == "0";
+            var endsWithDash = lastLabel.EndsWith("-", StringComparison.Ordinal);
+
+            if (endsWithZero || endsWithDash)
             {
                 var fixedReleaseLabel = string.Empty;
 
-                if (lastLabel.EndsWith("--", StringComparison.Ordinal))
-                {
-                    // For labels such as rc1-* an additional - is added by nuget
-                    fixedReleaseLabel = lastLabel.Substring(0, lastLabel.Length - 2);
-                }
-                else
+                if (endsWithDash)
                 {
-                    // Remove the - for 1.0.0-* (1.0.0--)
-                    fixedReleaseLabel = lastLabel.Substring(0, lastLabel.Length - 1);
+                    if (lastLabel.EndsWith("--", StringComparison.Ordinal))
```

Comment:
Normally I'd agree, but here the default behavior is to do nothing. Two special cases exists where extra truncation is needed.